cmd/cluster-capacity/app/options/options.go

```diff
@@ -117,68 +124,111 @@ func (s *ClusterCapacityConfig) ParseAdditionalSchedulerConfigs() error {
-		}
-		spec = response.Body
-	} else {
-		filename, _ := filepath.Abs(s.Options.PodSpecFile)
-		spec, err = os.Open(filename)
-		if err != nil {
-			return fmt.Errorf("Failed to open config file: %v", err)
-		}
-	}
+func newFakeClient() resource.ClientMapper {
+	return resource.ClientMapperFunc(func(*meta.RESTMapping) (resource.RESTClient, error) {
+		return &fake.RESTClient{}, nil
+	})
+}
 
-	decoder := yaml.NewYAMLOrJSONDecoder(spec, 4096)
-	versionedPod := &v1.Pod{}
-	err = decoder.Decode(versionedPod)
-	if err != nil {
-		return fmt.Errorf("Failed to decode config file: %v", err)
+func getPodsLabelSet(template *v1.PodTemplateSpec) labels.Set {
+	desiredLabels := make(labels.Set)
+	for k, v := range template.Labels {
+		desiredLabels[k] = v
 	}
+	return desiredLabels
+}
 
-	if versionedPod.ObjectMeta.Namespace == "" {
-		versionedPod.ObjectMeta.Namespace = "default"
+// inspired by:
+// https://github.com/kubernetes/kubernetes/blob/release-1.6/pkg/controller/controller_utils.go#L512
+func GetPodFromTemplate(template *v1.PodTemplateSpec, parentObject runtime.Object) (*v1.Pod, error) {
+	desiredLabels := getPodsLabelSet(template)
+	accessor, err := meta.Accessor(parentObject)
+	if err != nil {
+		return nil, fmt.Errorf("parentObject does not have ObjectMeta, %v", err)
 	}
 
-	// hardcoded from kube api defaults and validation
-	// TODO: rewrite when object validation gets more available for non kubectl approaches in kube
-	if versionedPod.Spec.DNSPolicy == "" {
-		versionedPod.Spec.DNSPolicy = v1.DNSClusterFirst
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: desiredLabels,
+			Name:   accessor.GetName(),
+		},
 	}
-	if versionedPod.Spec.RestartPolicy == "" {
-		versionedPod.Spec.RestartPolicy = v1.RestartPolicyAlways
+	clone, err := api.Scheme.DeepCopy(&template.Spec)
+	if err != nil {
+		return nil, err
 	}
+	pod.Spec = *clone.(*v1.PodSpec)
+	return pod, nil
+}
 
-	for i := range versionedPod.Spec.Containers {
-		if versionedPod.Spec.Containers[i].TerminationMessagePolicy == "" {
-			versionedPod.Spec.Containers[i].TerminationMessagePolicy = v1.TerminationMessageFallbackToLogsOnError
+func (s *ClusterCapacityConfig) ParseAPISpec() error {
+	var err error
+
+	r := resource.NewBuilder(api.Registry.RESTMapper(), api.Scheme, newFakeClient(), api.Codecs.UniversalDecoder(v1.SchemeGroupVersion)).
+		FilenameParam(false, &resource.FilenameOptions{Filenames: []string{s.Options.PodSpecFile}, Recursive: false}).
+		ContinueOnError().
+		Flatten().
+		Do()
+
+	versionedPods := make([]framework.SimulatedPod, 0)
+	err = r.Visit(func(info *resource.Info, err error) error {
+		switch info.Object.(type) {
+		// TODO: Support Deployment and ReplicaSet as well
+		case *v1.Pod:
+			pod := framework.SimulatedPod{Pod: info.Object.(*v1.Pod), Replicas: 1}
+			versionedPods = append(versionedPods, pod)
+		case *v1.ReplicationController:
```

Comment:
@davidopp agree, but could we just add support for `ReplicaSet`? If I'm not mistaken `ReplicaSet` is still beta and we still have some `ReplicationController` resources